cachito/workers/pkg_managers/general_js.py

```diff
@@ -44,6 +51,77 @@
 log = logging.getLogger(__name__)
 
 
+def parse_dependency(proxy_repo_url, dep_identifier):
+    """
+    Parse the dependecy to return the proxied download URL and the file name.
+
+    :param str proxy_repo_url: The Nexus proxy repository URL to use as the registry
+    :param str dep_identifier:
+    :return: proxied_url for the download and the tarball_name.
+    :rtype: str, str
+    """
+    full_pkg_name, version = dep_identifier.rsplit("@", 1)
+    if "/" in full_pkg_name:
+        short_pkg_name = full_pkg_name.rsplit("/", 1)[1]
+    else:
+        short_pkg_name = full_pkg_name
+
+    proxied_url = f"{proxy_repo_url}{full_pkg_name}/-/{short_pkg_name}-{version}.tgz"
+    tarball_name = f"{full_pkg_name}-{version}.tgz"
+    # asserts position at start of the string matches the character "@"
+    tarball_name = re.sub(r"^@", "", tarball_name)
+    # matches the character "/"
+    tarball_name = re.sub(r"\/", "-", tarball_name)
```

Comment:
Would it be simpler like this?
```suggestion
    tarball_name = f"{full_pkg_name}-{version}.tgz".removeprefix("@").replace("/", "-")
```
And another suggestion: could you add a comment with a link to the npm code (mentioned in another comment) that explains why we do these replacements?